libcaja-private/caja-autorun.c

```diff
@@ -753,15 +753,16 @@ static gboolean
 is_shift_pressed (void)
 {
     gboolean ret;
+    GdkDisplay *display;
     XkbStateRec state;
     Bool status;
 
     ret = FALSE;
 
-    gdk_error_trap_push ();
+    gdk_x11_display_error_trap_push (display);
     status = XkbGetState (GDK_DISPLAY_XDISPLAY (gdk_display_get_default ()),
     			  XkbUseCoreKbd, &state);
-    gdk_error_trap_pop_ignored ();
+    gdk_x11_display_error_trap_pop_ignored (display);
 
```

Comment:
here the variable 'display' isn't defined, maybe this line is needed before?

display = gdk_display_get_default ();

Reply:
i agree, just updated the PR